### PR TITLE
GEODE-2508: Initial work on new approach to generic lib naming.

### PR DESCRIPTION
**Pull request by echobravopapa**
Keeping the changes to CppCacheLibrary.cpp - did not, yet, find a CMake way to do it more seamlessly.

### CODE REVIEW

**Comment by pivotal-jbarrett** [comment]
@PivotalSarge yes.

File: src/cppcache/src/CMakeLists.txt [hunk 1]
```diff
@@ -125,7 +125,7 @@ target_include_directories(apache-geode
     $<TARGET_PROPERTY:_apache-geode,INTERFACE_INCLUDE_DIRECTORIES>
 )
 add_dependencies(client-libraries apache-geode)
-set_target_properties(apache-geode PROPERTIES PUBLIC_HEADER "${PUBLIC_HEADERS}")
+set_target_properties(apache-geode PROPERTIES PUBLIC_HEADER "${PUBLIC_HEADERS}" OUTPUT_NAME ${PRODUCT_LIB_NAME} )
```
Comment (pivotal-jbarrett) [inline src/cppcache/src/CMakeLists.txt]: I would only suggest avoiding long lines in the CMake files by putting each property on a new line.

```
set_target_properties(apache-geode PROPERTIES
    PUBLIC_HEADER "${PUBLIC_HEADERS}"
    OUTPUT_NAME ${PRODUCT_LIB_NAME})
```

File: src/cppcache/src/CppCacheLibrary.cpp [hunk 2]
```diff
@@ -127,15 +127,18 @@ std::string CppCacheLibrary::getProductLibDir() {
   for (int i = 0; i < PATH_MAX && path[i] != 0; i++) {
     path[i] = ::tolower(path[i]);
   }
-  dllNamePtr = strstr(path, "apache-geode.dll");
+  std::string cppName = PRODUCT_LIB_NAME;
+  cppName += ".dll";
+  std::string dotNetName = PRODUCT_DLL_NAME;
+  dotNetName += ".dll";
+  dllNamePtr = strstr(path, cppName.c_str());
```
Comment (dgkimura) [inline src/cppcache/src/CppCacheLibrary.cpp]: Nit: `strstr` seems old-school.  If you turn `path` into a `std::string` then you can use `std::basic_string::find` and do something like.

```cpp
if (pathAsString.find(cppName) == std::string::npos)
{
    // ...
}
```

File: src/cppcache/src/CMakeLists.txt (hunk 1 above)
Comment (echobravopapa) [inline src/cppcache/src/CMakeLists.txt]: will adjust that.  cmake did not like duplicate calls to set_target_properites

File: src/cppcache/src/CppCacheLibrary.cpp (hunk 2 above)
Comment (echobravopapa) [inline src/cppcache/src/CppCacheLibrary.cpp]: fair nit 👍 will change it

**Pull request closed without merging**